contrib/golang/filters/http/source/processor_state.cc

```diff
@@ -47,13 +47,18 @@ bool BufferList::checkExisting(Buffer::Instance* data) {
 };
 
 // headers_ should set to nullptr when return true.
-bool ProcessorState::handleHeaderGolangStatus(const GolangStatus status) {
+bool ProcessorState::handleHeaderGolangStatus(GolangStatus status) {
   ENVOY_LOG(debug, "golang filter handle header status, state: {}, phase: {}, status: {}",
             stateStr(), phaseStr(), int(status));
 
   ASSERT(state_ == FilterState::ProcessingHeader);
   bool done = false;
 
+  if (do_end_stream_ &&
+      (status == GolangStatus::StopAndBuffer || status == GolangStatus::StopAndBufferWatermark)) {
+    status = GolangStatus::Continue;
```

Comment:
maybe it worth a warning log? since it is a unexpected status, there might be something wrong in the Golang plugin.

Reply:
If this is broken it's going to log spam at high right. I would use ENVOY_BUG or one of the rate limiting logging messages.

Reply:
This path is triggered when the user provides a wrong status code from their plugin, so using ENVOY_BUG may be misleading, as it is user's BUG.

@mattklein123 
Is there any rate limiting logging messages mechanism that we can take advantage of?

Reply:
@spacewander Seems there isn't a  rate limiting logging messages mechanism yet.

or maybe we can check the status in the Golang filter filter manager, and it will panic when status is unexpected, instead of ignore it?
and panic means the request will be terminated with 500 status, and do not affect other requests.

Reply:
> @spacewander Seems there isn't a rate limiting logging messages mechanism yet.
> 
> or maybe we can check the status in the Golang filter filter manager, and it will panic when status is unexpected, instead of ignore it? and panic means the request will be terminated with 500 status, and do not affect other requests.

LGTM. I will do it later.